File: PromoStudio.Common/Models/Storyboard.cs
using System;
using System.Collections.Generic;
using PromoStudio.Common.Enumerations;

namespace PromoStudio.Common.Models
{
    public class Storyboard
    {
        private List<StoryboardItem> _storyboardItems = new List<StoryboardItem>();

        public int pk_StoryboardId { get; set; }
        public sbyte fk_StoryboardStatusId { get; set; }
        public int? fk_OrganizationId { get; set; }
        public int? fk_VerticalId { get; set; }
        public int? fk_AudioScriptTemplateId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public long VimeoVideoId { get; set; }
        public string VimeoThumbnailUrl { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime? DateUpdated { get; set; }

        public AudioScriptTemplate AudioScriptTemplate { get; set; }

        public List<StoryboardItem> Items
        {
            get { return _storyboardItems; }
            set { _storyboardItems = value; }
        }

        public StoryboardStatus Status
        {
            get { return (StoryboardStatus) fk_StoryboardStatusId; }
            set { fk_StoryboardStatusId = (sbyte) value; }
        }

        public dynamic ToPoco()
        {
            return new
            {
                pk_StoryboardId,
                fk_StoryboardStatusId,
                fk_OrganizationId,
                fk_VerticalId,
                fk_AudioScriptTemplateId,
                Name,
                Description,
                VimeoVideoId,
                VimeoThumbnailUrl,
                DateCreated,
                DateUpdated
            };
        }
    }
}

File: PromoStudio.Web/Scripts/models/storyboard.js
/// <reference path="../vsdoc/require.js" />
/// <reference path="../vsdoc/knockout-3.0.0.debug.js" />
/// <reference path="audioScriptTemplate.js" />
/// <reference path="storyboardItem.js" />
/// <reference path="../ps/extensions.js" />

"use strict";

define([
    "models/storyboardItem",
    "models/audioScriptTemplate",
    "knockout",
    "ps/extensions"
    ], function (
        storyboardItem,
        audioScriptTemplate,
        ko) {
    function ctor (data) {
        var self = this;
        data = data || {};

        self.pk_StoryboardId = ko.observable(data.pk_StoryboardId || null);
        self.fk_StoryboardStatusId = ko.observable(data.fk_StoryboardStatusId || null);
        self.fk_OrganizationId = ko.observable(data.fk_OrganizationId || null);
        self.fk_VerticalId = ko.observable(data.fk_VerticalId || null);
        self.fk_AudioScriptTemplateId = ko.observable(data.fk_AudioScriptTemplateId || null);
        self.Name = ko.observable(data.Name || null);
        self.Description = ko.observable(data.Description || null);
        self.VimeoVideoId = ko.observable(data.VimeoVideoId || null);
        self.VimeoThumbnailUrl = ko.observable(data.VimeoThumbnailUrl || null);

        self.Items = ko.observableArray([]);

        self.AudioScriptTemplate = ko.observable(null);

        self.LoadItems = function (audioScript, items) {
            var i, item;
            items = items || [];

            for (i = 0; i < items.length; i++) {
                item = items[i];
                items[i] = new storyboardItem(self, item);
            }

            self.Items(items);
            
            if (audioScript) {
                self.AudioScriptTemplate(new audioScriptTemplate(audioScript));
            }
        };
        self.LoadItems(data.AudioScriptTemplate, data.Items);
    }

    ctor.prototype.toJSON = function () {
        var copy = ko.toJS(this);

        // remove any unneeded properties
        delete copy.AudioScriptTemplate;

        return copy;
    };
    return ctor;
});
